R/data-combined.R

```diff
@@ -0,0 +1,860 @@
+      # - extract group map
+      # - extract dataset names
+
+      # styler: off
+      private$.dataCombined <- private$.updateDF(private$.dataCombined, private$.dataSetToDF(dataSets, names, groups))
+      private$.dataCombined <- private$.extractXYData(private$.dataCombined)
+      private$.groupMap     <- private$.extractGroupMap(private$.dataCombined)
+      private$.names        <- private$.extractNames(private$.dataCombined)
+      # styler: on
+
+      # set up data transformations
+      self$setDataTransformations(names)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @description
+    #' Add simulated data.
+    #' @return `DataCombined` object containing simulated data.
+
+    addSimulationResults = function(simulationResults,
+                                    quantitiesOrPaths = NULL,
+                                    population = NULL,
+                                    individualIds = NULL,
+                                    names = NULL,
+                                    groups = NULL) {
+      # list or a vector of `SimulationResults` class instances is not allowed
+      if (is.vector(simulationResults)) {
+        stop(messages$errorOnlyOneSupported())
+      }
+
+      # validate the object type (`NULL` elements are not accepted)
+      validateIsOfType(simulationResults, SimulationResults, nullAllowed = FALSE)
+
+      # validate argument type and length
+      groups <- validateVectorArgs(groups, length(quantitiesOrPaths %||% simulationResults$allQuantityPaths), type = "character")
+      names <- validateVectorArgs(names, length(quantitiesOrPaths %||% simulationResults$allQuantityPaths), type = "character")
+
+      # Update private fields for the new setter call
+      #
+      # - extract dataframe and append it to the combined dataframe
+      # - extract original XY data into separate columns
+      # - extract group map
+      # - extract dataset names
+
+      # styler: off
+      private$.dataCombined <- private$.updateDF(private$.dataCombined,
+        private$.simResultsToDF(
+          simulationResults = simulationResults,
+          quantitiesOrPaths = quantitiesOrPaths,
+          population        = population,
+          individualIds     = individualIds,
+          names             = names,
+          groups            = groups
+        )
+      )
+      private$.dataCombined <- private$.extractXYData(private$.dataCombined)
+      private$.groupMap     <- private$.extractGroupMap(private$.dataCombined)
+      private$.names        <- private$.extractNames(private$.dataCombined)
+      # styler: on
+
+      # set up data transformations
+      self$setDataTransformations(names)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @description
+    #'
+    #' Transform raw data with required offsets and scale factors.
+    #'
+    #' @param xOffsets,yOffsets,xScaleFactors,yScaleFactors Either a numeric
```

Comment:
Add a description how transformations are applied. I.e. for x/yValues, first the offset is added and then the value is multiplied by factor; for yErrorValues only multiplying by factor.